integration/ctr/oci_hooks_tests.bats

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bats
+
+IMAGE="${IMAGE:-docker.io/library/busybox:latest}"
+CONTAINER_NAME="${CONTAINER_NAME:-kataocihookstest}"
+CTR_RUNTIME="${CTR_RUNTIME:-io.containerd.kata.v2}"
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+readonly KATA_CONFIG="${KATA_CONFIG:-/etc/kata-containers/configuration.toml}"
+readonly KATA_CONFIG_BACKUP="$KATA_CONFIG.backup"
+readonly DEFAULT_KATA_CONFIG="/usr/share/defaults/kata-containers/configuration.toml"
+
+function teardown() {
+	sudo rm -rf $HOST_TMP_TEST_DIR
+	[ -f "$KATA_CONFIG_BACKUP" ] && sudo mv "$KATA_CONFIG_BACKUP" "$KATA_CONFIG" || \
+		sudo cp "$DEFAULT_KATA_CONFIG" "$KATA_CONFIG"
+}
+
+function setup() {
+	HOST_TMP_TEST_DIR="$(mktemp -d)"
+	sudo mkdir -p $(dirname "${KATA_CONFIG}")
+	[ -f "$KATA_CONFIG" ] && sudo cp "$KATA_CONFIG" "$KATA_CONFIG_BACKUP" || \
+		sudo cp "$DEFAULT_KATA_CONFIG" "$KATA_CONFIG"
+}
+
+
+@test "Ensure poststart OCI hooks work" {
+	local tmp_guest_img_path="$HOST_TMP_TEST_DIR/kata-containers.img"
+	local tmp_hooks_filepath="$HOST_TMP_TEST_DIR/hooks.sh"
+	local tmp_mount_point="$HOST_TMP_TEST_DIR/mnt"
+
+	# leveraging an oci poststart hook so that we can inspect the file after the container is up and running.
+	local tmp_oci_hooks_dir="$tmp_mount_point/usr/share/oci/hooks/poststart"
+
+	# add file with greppable phrase to file that is shared by guest vm and container.
+	printf "#!/bin/sh\necho \"foo\" > /run/kata-containers/%s/rootfs/tmp/test-oci-hooks-file\n" "$CONTAINER_NAME" > $tmp_hooks_filepath
+	chmod +x $tmp_hooks_filepath
+
+	# prepare guest image with hooks
+	mkdir $tmp_mount_point
+	cp /usr/share/kata-containers/kata-containers.img $tmp_guest_img_path
+	sudo mount -o loop,offset=$((512*6144)) $tmp_guest_img_path $tmp_mount_point
+	sudo mkdir -p $tmp_oci_hooks_dir
+	sudo cp $tmp_hooks_filepath $tmp_oci_hooks_dir
+	sudo umount $tmp_mount_point
+
+	# add guest image to config file and enable oci hooks.
+	mkdir -p /etc/kata-containers
+	sudo sed -i -r "s#^(image =).*#\1 \"$tmp_guest_img_path\"#" "$KATA_CONFIG"
+	sudo sed -i '/^#guest_hook_path /s/^#//' "$KATA_CONFIG"
+
+	# pull an image with ctr to run and test for oci hooks
+	sudo ctr image pull "$IMAGE"
+	[ $? != 0 ] && die "Unable to get image $IMAGE"
+
+	run sudo ctr run --rm --runtime="$CTR_RUNTIME" "$IMAGE" "$CONTAINER_NAME" sh -c "grep foo /tmp/test-oci-hooks-file"
```

Comment:
Could you create a variable for `/tmp/test-oci-hooks-file` as it's used multiple times.